src/traits.jl

```diff
@@ -58,3 +58,33 @@ Return the [`OutputTrait`] of the [`Model`](@ref) or [`Template`](@ref).
 """
 output_type(::T) where T = output_type(T)
 output_type(T::Type) = throw(MethodError(output_type, (T,)))  # to prevent recursion
+
+"""
+    InputTrait
+
+The `InputTrait` specifies if the model supports point or distribution inputs to predict,
```

Comment:
Can we handle models that accept either, maybe something like `PointOrDistributionInput`?